src/Facets/ChainflipFacet.sol

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { LibDiamond } from "../Libraries/LibDiamond.sol";
+import { LibAsset } from "../Libraries/LibAsset.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { SwapperV2 } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+import { IChainflipVault } from "../Interfaces/IChainflip.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { console } from "forge-std/console.sol";
+
+/// @title Chainflip Facet
+/// @author LI.FI (https://li.fi)
+/// @notice Allows bridging assets via Chainflip
+/// @custom:version 1.0.0
+contract ChainflipFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    /// Events ///
+    event BridgeToNonEVMChain(
+        bytes32 indexed transactionId,
+        uint256 indexed destinationChainId,
+        bytes32 receiver
+    );
+
+    /// Errors ///
+    error EmptyNonEvmAddress();
+    address public immutable chainflipVault;
+
+    uint256 private constant CHAIN_ID_ETHEREUM = 1;
+    uint256 private constant CHAIN_ID_ARBITRUM = 42161;
+    uint256 private constant CHAIN_ID_SOLANA = 1151111081099710;
+    uint256 private constant CHAIN_ID_BITCOIN = 20000000000001;
+
+    uint32 private constant CHAINFLIP_ID_ETHEREUM = 1;
+    uint32 private constant CHAINFLIP_ID_ARBITRUM = 4;
+    uint32 private constant CHAINFLIP_ID_SOLANA = 5;
+    uint32 private constant CHAINFLIP_ID_BITCOIN = 3;
+
+    /// Types ///
+
+    /// @dev Parameters specific to Chainflip bridge
+    /// @param dstToken Token to be received on the destination chain (uint32)
+    /// @param cfParameters Additional metadata for future features (currently unused)
+    struct ChainflipData {
+        uint32 dstToken;
+        bytes32 nonEvmAddress;
+        bytes cfParameters;
+    }
+
+    /// Constructor ///
+
+    /// @notice Constructor for the contract.
+    /// @param _chainflipVault Address of the Chainflip vault contract
+    constructor(address _chainflipVault) {
+        chainflipVault = _chainflipVault;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via Chainflip
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _chainflipData Data specific to Chainflip
+    function startBridgeTokensViaChainflip(
+        ILiFi.BridgeData memory _bridgeData,
+        ChainflipData calldata _chainflipData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
```

Comment:
you said the facet supports destination calls but then we have this modifier...that doesnt fit together imho
(same in function below)